modules/apps/portal/portal-tika/src/main/java/com/liferay/portal/tika/internal/util/TikaConfigHelper.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.tika.internal.util;
+
+import com.liferay.petra.io.StreamUtil;
+import com.liferay.petra.io.unsync.UnsyncBufferedInputStream;
+import com.liferay.petra.io.unsync.UnsyncByteArrayInputStream;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.exception.SystemException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.PortalClassLoaderUtil;
+import com.liferay.portal.tika.internal.configuration.TikaConfiguration;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.util.Map;
+
+import org.apache.tika.config.TikaConfig;
+import org.apache.tika.detect.Detector;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.parser.Parser;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * @author Shuyang Zhou
+ * @author Jorge Díaz
+ */
+@Component(
+	configurationPid = "com.liferay.portal.tika.internal.configuration.TikaConfiguration",
+	service = TikaConfigHelper.class
+)
+public class TikaConfigHelper {
+
+	public TikaConfig getTikaConfig() {
+		return _tikaConfig;
+	}
+
+	public String getTikaConfigXml() throws IOException {
+		InputStream inputStream = TikaConfigHelper.class.getResourceAsStream(
+			_tikaConfiguration.tikaConfigXml());
+
+		if (inputStream == null) {
+			ClassLoader classLoader = PortalClassLoaderUtil.getClassLoader();
+
+			inputStream = classLoader.getResourceAsStream(
+				_tikaConfiguration.tikaConfigXml());
+		}
+
+		return StreamUtil.toString(inputStream);
+	}
+
+	public Parser getTikaParser() {
+		return _parser;
+	}
+
+	public boolean isTextExtractionForkProcessEnabled(InputStream inputStream)
```

Comment:
@jorgediaz-lr Only one is using this method, please inline it into the only usage

Reply:
I have moved this method to `TextExtractorImpl._isTextExtractionForkProcessEnabled(InputStream)`, see new commit https://github.com/liferay-core-infra/liferay-portal/commit/c20aac94c34b33acda8ec3903ceefcbab0ac91b5

I have to maintain a private method because I have created a test to check that this functionality is working correctly, see my new test in commit https://github.com/liferay-core-infra/liferay-portal/commit/70781afc8945d64a0e6126796023d9448449c40f